component/src/main/java/org/wso2/extension/siddhi/execution/json/function/InsertToJSONFunctionExtension.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c)  2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.execution.json.function;
+
+import com.google.gson.Gson;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.InvalidModificationException;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.PathNotFoundException;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.annotation.Example;
+import org.wso2.siddhi.annotation.Extension;
+import org.wso2.siddhi.annotation.Parameter;
+import org.wso2.siddhi.annotation.ReturnAttribute;
+import org.wso2.siddhi.annotation.util.DataType;
+import org.wso2.siddhi.core.config.SiddhiAppContext;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * This class provides implementation for inserting values to the given json using the path specified.
+ */
+@Extension(
+        name = "insertIntoJson",
```

Comment:
Can we set both elements and attributes? can we set bool, int, and others ? 
Shall we use a name like setElement()

Why return null when the path is wrong? 
It should be  "it will return the original json. Otherwise, it will return the updated json."

Reply:
We can do the following with the function,

- add new attributes to existing JSON components
- modify existing attributes in JSON components
- add json elements to JSON arrays

Moreover, we can set bool, string and number types.